spec.md

```diff
@@ -535,9 +540,54 @@ message PluginCapability {
     Type type = 1;
   }
 
+  message VolumeExpansion {
+    enum Type {
+      UNKNOWN = 0;
+
+      // ONLINE indicates that volumes may be expanded when published to
+      // a node. When a Plugin implements this capability it MUST
+      // implement either the EXPAND_VOLUME controller capability or the
+      // EXPAND_VOLUME node capability or both. When a plugin supports
+      // ONLINE volume expansion and also has the EXPAND_VOLUME
+      // controller capability then the plugin MUST support expansion of
+      // volumes currently published and available on a node. When a
+      // plugin supports ONLINE volume expansion and also has the
+      // EXPAND_VOLUME node capability then the plugin MAY support
+      // expansion of node-published volume via NodeExpandVolume.
+      //
+      // Example 1: Given a shared filesystem volume (e.g. GlusterFs),
+      //   the Plugin may set the ONLINE volume expansion capability and
+      //   implement ControllerExpandVolume but not NodeExpandVolume.
+      //
+      // Example 2: Given a block storage volume type (e.g. EBS), the
+      //   Plugin may set the ONLINE volume expansion capability and
+      //   implement both ControllerExpandVolume and NodeExpandVolume.
+      //
+      // Example 3: Given a Plugin that supports volume expansion only
+      //   on upon a node, the Plugin may set the ONLINE volume
+      //   expansion capability and implement NodeExpandVolume but not
+      //   ControllerExpandVolume.
+      ONLINE = 1;
+
+      // OFFLINE indicates that volumes currently published and
+      // available on a node MAY NOT be expanded. When a plugin support
+      // OFFLINE volume expansion it MUST implement either the
+      // EXPAND_VOLUME controller capability or the EXPAND_VOLUME
```

Comment:
EXPAND_VOLUME node capability doesn't make sense for OFFLINE?

Reply:
or .. we just don't want to support controller-volume-expand w/ OFFLINE but we *do* want to support node-volume-expand for OFFLINE.

we should clarify this because the paragraph is confusing

Reply:
IIRC according to a meeting we had a while ago, it seems to me that the `EXPAND_VOLUME` controller capability is necessary, but the `EXPAND_VOLUME` node capability is not, and the CO won't call `NodeExpandVolume` to expand a volume offline. It might be worthy mentioning that the plugin can choose to complete the node-side work (e.g. filesystem changes) in the next `NodeStageVolume` or `NodePublishVolume` call.

Reply:
I have updated the wording to accurately reflect that OFFLINE here only applies to `ControllerExpandVolume` . 

@chhsia0 the wording of `NodeExpandVolume` already says that it only supports expansion of published/staged volumes. 

>IRC according to a meeting we had a while ago, it seems to me that the EXPAND_VOLUME controller capability is necessary, but the EXPAND_VOLUME node capability is not, 

Are you saying that, every plugin that supports volume expansion of some sort MUST implement `EXPAND_VOLUME` controller capability? I do not recall we agreed to that. I do remember that, we agreed that - capability will ONLY apply to ControllerExpandVolume RPC call and not NodeExpandVolume call.

Reply:
No this is not what I mean. See below.